src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+        angle = valueInContext(self.angle, context)
+        return SectorRegion(center, radius, heading, angle,
+                            name=self.name, resolution=self.resolution)
+
+    def containsPoint(self, point):
+        point = toVector(point)
+
+        if point.z != self.z:
+            return False
+
+        if not pointIsInCone(tuple(point), tuple(self.center), self.heading, self.angle):
+            return False
+
+        return point.distanceTo(self.center) <= self.radius
+
+    def uniformPointInner(self):
+        x, y, z = self.center
+        heading, angle, maxDist = self.heading, self.angle, self.radius
+        r = random.triangular(0, maxDist, maxDist)
+        ha = angle / 2.0
+        t = random.uniform(-ha, ha) + (heading + (math.pi / 2))
+        pt = Vector(x + (r * cos(t)), y + (r * sin(t)), z)
+        return self.orient(pt)
+
+    def __repr__(self):
+        return f'SectorRegion({self.center!r},{self.radius!r},{self.heading!r},{self.angle!r})'
+
+class RectangularRegion(PolygonalRegion):
+    """A rectangular region with a possibly-random position, heading, and size.
+
+    Args:
+        position (`Vector`): center of the rectangle.
+        heading (float): the heading of the ``length`` axis of the rectangle.
+        width (float): width of the rectangle.
+        length (float): length of the rectangle.
+        name (str; optional): name for debugging.
+    """
+    def __init__(self, position, heading, width, length, name=None):
+        self.position = toVector(position, "position of RectangularRegion not a vector")
+        self.heading = toScalar(heading, "heading of RectangularRegion not a scalar")
+        self.width = toScalar(width, "width of RectangularRegion not a scalar")
+        self.length = toScalar(length, "length of RectangularRegion not a scalar")
+
+        deps = [self.position, self.heading, self.width, self.length]
+
+        self.hw = hw = width / 2
+        self.hl = hl = length / 2
+        self.radius = hypot(hw, hl)     # circumcircle; for collision detection
+        self.corners = tuple(self.position.offsetRotated(heading, Vector(*offset))
+            for offset in ((hw, hl), (-hw, hl), (-hw, -hl), (hw, -hl)))
+        self.circumcircle = (self.position, self.radius)
+
+        if any(needsSampling(dep) or needsLazyEvaluation(dep) for dep in deps):
+            # Center and radius aren't fixed, so we'll wait to pass a polygon
+            super().__init__(polygon=None, name=name, additionalDeps=deps)
+            return
+
+        polygons = self._make_rectangle_polygons()
+        super().__init__(polygon=polygons, z=self.position.z, name=name, additionalDeps=deps)
+
+    def _make_rectangle_polygons(self):
+        position, heading, hw, hl = self.position, self.heading, self.hw, self.hl
+        assert not any(isLazy(c) for c in (position, heading, hw, hl))
+        corners = _RotatedRectangle.makeCorners(position.x, position.y, heading, hw, hl)
```

Comment:
Let's get rid of `_RotatedRectangle` -- just grab the code from `Object._boundingPolygon`.